### ISSUE TITLE
`render_item` param in `context.configure` may return other than boolean

### ISSUE THREAD

**Issue by koki-sato** [sqlalchemy/alembic#1175]
**Describe the bug**

[`render_item`](https://github.com/sqlalchemy/alembic/blob/rel_1_9_3/alembic/context.pyi#L99) param in `context.configure` is defined as `Optional[Callable[... , bool]]`, but may actually return string or None.

https://alembic.sqlalchemy.org/en/latest/autogenerate.html#affecting-the-rendering-of-types-themselves

```python
def render_item(type_, obj, autogen_context):
    """Apply custom rendering for selected items."""

    if type_ == 'type' and isinstance(obj, MySpecialType):
        return "mypackage.%r" % obj

    # default rendering for other objects
    return False

def run_migrations_online():
    # ...

    context.configure(
                connection=connection,
                target_metadata=target_metadata,
                render_item=render_item,
                # ...
                )

    # ...
```

I get an error with type checking tools like Pylance.

<img width="1200" alt="image" src="https://user-images.githubusercontent.com/12211516/218656864-1b716456-1973-4aa5-9c4d-3b727043645d.png">

**Expected behavior**

Type Hints are defined correctly.

**Versions.**
 - OS: 
 - Python:
 - Alembic:
 - SQLAlchemy:
 - Database:
 - DBAPI:

**Additional context**

**Have a nice day!**

**Comment by CaselIT** [comment]
Hi,

Thanks for reporting. We should also check if there are tests for this behaviour

**Comment by CaselIT** [comment]
This was fixed in 4c5f80b for https://github.com/sqlalchemy/alembic/issues/1221
